### PR TITLE
Added Basal Icicle mode; store basal mode in localstorage

### PR DESCRIPTION
**Pull request by jasoncalabrese**
screen shot 2015-10-25 at 12 26 06 am

screen shot 2015-10-25 at 12 27 38 am

### CODE REVIEW

File: lib/plugins/basalprofile.js [hunk 1]
```diff
@@ -6,18 +6,37 @@ function init() {
...
+
+    var mode = pluginBase.getFromStorage(basalDisplayAttr, 'none');
+    basalSwitch.val(mode);
+    window.Nightscout.client.chart.basals.attr('display', toDisplayAttr(mode));
+
+    basalSwitch.change(function switchChart ( ) {
+      var mode = basalSwitch.val();
+      window.Nightscout.client.chart.basals.attr('display', toDisplayAttr(mode));
+      if ('none' !== toDisplayAttr(mode)) {
+        window.Nightscout.client.chart.setBasalMode(mode)
```
Comment (jasoncalabrese) [inline lib/plugins/basalprofile.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1181988562/issues/source?bid=2452379&fileBranchId=2921640#l34)

**Comment by bewest** [comment]
There's some regression in dev (before this patch) but before `5dc49692432229892613fb022c9aa176eb62f5ff` in `wip/increase-upload-limit` which is preventing temp basals from being displayed.

**Comment by bewest** [comment]
Comments are that on laptop/desktop the selector is extremely small.  Otherwise it's pretty awesome to have multiple choices here.

**Comment by jasoncalabrese** [comment]
Think we need to open a panel instead of trying to use a select for this, maybe something simple like we do for snoozing alarms.

**Comment by bewest** [comment]
image

**Comment by bewest** [comment]
Looks great:
image

Working great:
image

**Comment by bewest** [comment]
Err, here's one with the right screenshot:

image

**Pull request merged**